chain-tx-enclave-next/mls/src/group.rs

```diff
@@ -174,19 +175,38 @@ impl GroupAux {
             .map(|plain| ProposalId(self.tree.cs.hash(&plain.get_encoding())))
             .collect();
         let mut updated_tree = self.tree.clone();
+        let mut updated_group_context = GroupContext {
+            tree_hash: updated_tree.compute_tree_hash(),
+            ..self.context.clone()
+        };
         let positions = updated_tree.update(&add_proposals, &[], &[]);
-        let commit_secret = vec![0; self.tree.cs.hash_len()];
+
+        // update path secrets
+        let mut leaf_secret = vec![0; self.tree.cs.hash_len()];
+        rand::thread_rng().fill_bytes(&mut leaf_secret);
+        let (path_nodes, parent_hash, commit_secret) =
+            updated_tree.encap(&updated_group_context.get_encoding(), leaf_secret);
+
+        // update keypackage's parent_hash extension
+        self.owned_kp
+            .keypackage
+            .payload
+            .put_extension(&ext::ParentHashExt(parent_hash));
+        self.owned_kp.update_signature();
+        updated_tree.set_my_package(self.owned_kp.keypackage.clone());
```

Comment:
> The generator of the Commit starts by using the HPKE secret key "leaf_hpke_secret" associated with the new leaf KeyPackage (see {{key-packages}}) to compute "path_secret[0]"

Reply:
init key is to be updated -- https://github.com/cisco/go-mls/blob/master/state.go#L370 ?

Reply:
> When used as InitKeys, KeyPackages are intended to be used only once and SHOULD NOT be reused except in case of last resort.

Reply:
done